addons/stock_account/tests/test_stockvaluationlayer.py

```diff
@@ -778,7 +778,9 @@ def test_standard_manual_to_auto_2(self):
             'property_stock_journal': self.stock_journal.id,
         })
 
-        self.product1.categ_id = cat2
+        # Test with context value of product.
+        self.product1.with_context(default_type='product').categ_id = cat2
+        self.assertEqual(self.product1.categ_id, cat2)
```

Comment:
i don't understand this commit at all?

Reply:
Actually issue is generating when stock quant is present and from product from view -> change the product category(AVCO).
Because in default context, type of product is passed to account.move and type field is also there. and type='product' raised issue in stock.move.
By writing another category , tried to check the case.

Reply:
okay, now i get it

``` python
# Try to change the product category with a `default_type` key in the context and
# check it doesn't break the account move generation.
self.product1.with_context(default_type='product').categ_id = cat2
self.assertEqual(self.product1.categ_id, cat2)
```